core-kotlin-modules/core-kotlin-7/src/main/kotlin/com/baeldung/optional/Optional.kt

```diff
@@ -0,0 +1,33 @@
+package com.baeldung.optional
+
+import java.util.Optional
+import kotlin.jvm.optionals.getOrDefault
+import kotlin.jvm.optionals.getOrElse
+import kotlin.jvm.optionals.getOrNull
+
+fun getOrNull() {
+    val optionalValue: Optional<String> = Optional.of("Hello, Baeldung!")
+    val value: String? = optionalValue.getOrNull()
+}
+
+fun getOrElse() {
+    val optionalValue: Optional<String> = Optional.empty()
+    val value: String = optionalValue.getOrElse { throw IllegalStateException("Value is missing") }
+    val value2: String = optionalValue.getOrElse { "Default Value" }
+}
+
+fun getOrDefault() {
+    val optionalValue: Optional<String> = Optional.empty()
+    val value1: String = optionalValue.getOrDefault("Default Value")
+}
+
+data class User(val name: String)
+
+fun getUsername(user: User?): String {
+    val optionalUser = Optional.ofNullable<User>(user)
+    return optionalUser.map(User::name).orElse("Unknown")
+}
+
+fun getUsernameKotlin(user: User?): String {
+    return user?.name ?: "Unknown"
+}
```

Comment:
Let's actually turn this example code into a unit test.  The code in the article is fine though. For this, since it's optional unwrapping and nullability, I think just asserting null vs non null is enough.